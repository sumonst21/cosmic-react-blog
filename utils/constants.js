const config = require('../config');

const BASE_URL = process.env.NODE_ENV === 'production'
  ? config.BASE_URL
  : `http://localhost:${process.env.PORT}`;

const API_URL = process.env.NODE_ENV === 'production'
  ? config.API_URL
  : `http://localhost:${process.env.PORT || 8080}/api`;

const THEMES = {
  DARK: 'dark',
  LIGHT: 'light',
};

export {
  API_URL,
  BASE_URL,
  THEMES,
};
