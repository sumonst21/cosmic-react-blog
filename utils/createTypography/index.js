/*
  Adapted from source:
  https://github.com/mui-org/material-ui/blob/303199d39b42a321d28347d8440d69166f872f27/packages/material-ui/src/styles/createTypography.js
*/
const round = value => Math.round(value * 1e5) / 1e5;

const createTypography = (htmlFontSize = 16, palette) => {
  const sansFontFamily = "'Montserrat', sans-serif";
  const serifFontFamily = "'Lora', serif";
  const fontSize = 16;
  const fontWeightLight = 300;
  const fontWeightRegular = 400;
  const fontWeightMedium = 500;

  const coef = fontSize / 14;
  const pxToRem = value => `${(value / htmlFontSize) * coef}rem`;

  return {
    pxToRem,
    round,
    fontSize,
    fontWeightLight,
    fontWeightRegular,
    fontWeightMedium,
    sansFontFamily,
    serifFontFamily,
    display4: {
      fontSize: pxToRem(45),
      fontWeight: fontWeightRegular,
      fontFamily: sansFontFamily,
      lineHeight: `${round(48 / 45)}em`,
      marginLeft: '-.02em',
      color: palette.text.primary,
    },
    display3: {
      fontSize: pxToRem(45),
      fontWeight: fontWeightRegular,
      fontFamily: sansFontFamily,
      lineHeight: `${round(48 / 45)}em`,
      marginLeft: '-.02em',
      color: palette.text.primary,
    },
    display2: {
      fontSize: pxToRem(18),
      fontWeight: fontWeightMedium,
      fontFamily: sansFontFamily,
      lineHeight: `${round(48 / 45)}em`,
      marginLeft: '-.02em',
      color: palette.text.primary,
    },
    display1: {
      fontSize: pxToRem(16),
      fontWeight: fontWeightMedium,
      fontFamily: sansFontFamily,
      lineHeight: `${round(41 / 34)}em`,
      color: palette.text.primary,
    },
    headline: {
      fontSize: pxToRem(20),
      fontWeight: fontWeightMedium,
      fontFamily: sansFontFamily,
      lineHeight: `${round(32.5 / 24)}em`,
      color: palette.text.primary,
    },
    title: {
      fontSize: pxToRem(14),
      fontWeight: fontWeightMedium,
      fontFamily: sansFontFamily,
      lineHeight: `${round(24.5 / 21)}em`,
      color: palette.text.primary,
    },
    subheading: {
      fontSize: pxToRem(14),
      fontWeight: fontWeightRegular,
      fontFamily: sansFontFamily,
      lineHeight: `${round(24 / 16)}em`,
      color: palette.text.secondary,
    },
    body2: {
      fontSize: pxToRem(14),
      fontWeight: fontWeightRegular,
      fontFamily: serifFontFamily,
      lineHeight: `${round(24 / 14)}em`,
      color: palette.text.primary,
    },
    body1: {
      fontSize: pxToRem(16),
      fontWeight: fontWeightRegular,
      fontFamily: serifFontFamily,
      lineHeight: `${round(24 / 14)}em`,
      color: palette.text.primary,
    },
    caption: {
      fontSize: pxToRem(12),
      fontWeight: fontWeightRegular,
      fontFamily: sansFontFamily,
      lineHeight: `${round(16.5 / 12)}em`,
      color: palette.text.secondary,
    },
    button: {
      fontSize: pxToRem(14),
      textTransform: 'uppercase',
      fontWeight: fontWeightMedium,
      fontFamily: sansFontFamily,
      color: palette.text.primary,
    },
  };
};

export default createTypography;
