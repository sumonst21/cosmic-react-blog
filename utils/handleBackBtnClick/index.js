import { Router } from '../../routes';

const handleBackBtnClick = () => {
  // window.previouslyLoaded is set to true the first time App is loaded.
  // /pages/_app.jsx
  if (window.previouslyLoaded) {
    return Router.back();
  }

  return Router.pushRoute('/');
};

export default handleBackBtnClick;
