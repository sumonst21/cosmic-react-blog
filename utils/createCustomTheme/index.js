/*
  Adapted from source:
  https://github.com/mui-org/material-ui/blob/303199d39b42a321d28347d8440d69166f872f27/packages/material-ui/src/styles/createMuiTheme.js
*/
import createTypography from '../createTypography';
import createBreakpoints from '../../node_modules/@material-ui/core/styles/createBreakpoints';
import createPalette from '../../node_modules/@material-ui/core/styles/createPalette';
import createMixins from '../../node_modules/@material-ui/core/styles/createMixins';
import shadows from '../../node_modules/@material-ui/core/styles/shadows';
import transitions from '../../node_modules/@material-ui/core/styles/transitions';
import zIndex from '../../node_modules/@material-ui/core/styles/zIndex';
import spacing from '../../node_modules/@material-ui/core/styles/spacing';

const createCustomTheme = ({ htmlFontSize, theme }) => {
  const palette = createPalette({
    primary: {
      light: '#52c7b8',
      main: '#009688',
      dark: '#00675b',
      contrastText: '#000',
    },
    secondary: {
      light: '#6ed9ff',
      main: '#28a8e0',
      dark: '#0079ae',
      contrastText: '#000',
    },
    type: theme,
  });
  const breakpoints = createBreakpoints({});
  const mixins = createMixins(breakpoints, spacing, {});
  const typography = createTypography(htmlFontSize, palette);

  const muiTheme = {
    breakpoints,
    direction: 'ltr',
    mixins,
    overrides: {},
    palette,
    props: {},
    shadows,
    typography,
    transitions,
    spacing,
    zIndex,
  };

  return muiTheme;
};

export default createCustomTheme;
