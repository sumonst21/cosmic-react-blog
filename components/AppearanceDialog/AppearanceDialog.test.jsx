import { DisconnectedAppearanceDialog } from '.';
import { THEMES } from '../../utils/constants';

describe('AppearanceDialog', () => {
  it('should render', () => {
    const component = mount((
      <DisconnectedAppearanceDialog
        dispatch={jest.fn()}
        onClose={jest.fn()}
        open={false}
        theme={THEMES.LIGHT}
      />
    ));
    return expect(toJson(component)).toMatchSnapshot();
  });
});
