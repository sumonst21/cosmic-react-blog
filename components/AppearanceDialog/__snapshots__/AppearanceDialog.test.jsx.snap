// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppearanceDialog should render 1`] = `
<WithStyles(AppearanceDialog)
  dispatch={[MockFunction]}
  onClose={[MockFunction]}
  open={false}
  theme="light"
>
  <AppearanceDialog
    classes={
      Object {
        "button": "AppearanceDialog-button-1",
        "decreaseFontSizeBtn": "AppearanceDialog-decreaseFontSizeBtn-3",
        "dialog": "AppearanceDialog-dialog-2",
        "fontSizeBtnsContainer": "AppearanceDialog-fontSizeBtnsContainer-4",
        "increaseFontSizeBtn": "AppearanceDialog-increaseFontSizeBtn-5",
        "label": "AppearanceDialog-label-6",
        "switchContainer": "AppearanceDialog-switchContainer-7",
        "themeBtnsContainer": "AppearanceDialog-themeBtnsContainer-8",
      }
    }
    dispatch={[MockFunction]}
    onClose={[MockFunction]}
    open={false}
    theme="light"
  >
    <WithStyles(Dialog)
      aria-labelledby="Appearance"
      classes={
        Object {
          "paper": "AppearanceDialog-dialog-2",
        }
      }
      onClose={[MockFunction]}
      open={false}
    >
      <Dialog
        TransitionComponent={[Function]}
        aria-labelledby="Appearance"
        classes={
          Object {
            "paper": "MuiDialog-paper-10 AppearanceDialog-dialog-2",
            "paperFullScreen": "MuiDialog-paperFullScreen-15",
            "paperFullWidth": "MuiDialog-paperFullWidth-14",
            "paperWidthMd": "MuiDialog-paperWidthMd-13",
            "paperWidthSm": "MuiDialog-paperWidthSm-12",
            "paperWidthXs": "MuiDialog-paperWidthXs-11",
            "root": "MuiDialog-root-9",
          }
        }
        disableBackdropClick={false}
        disableEscapeKeyDown={false}
        fullScreen={false}
        fullWidth={false}
        maxWidth="sm"
        onClose={[MockFunction]}
        open={false}
        transitionDuration={
          Object {
            "enter": 225,
            "exit": 195,
          }
        }
      >
        <WithStyles(Modal)
          BackdropProps={
            Object {
              "transitionDuration": Object {
                "enter": 225,
                "exit": 195,
              },
            }
          }
          aria-labelledby="Appearance"
          className="MuiDialog-root-9"
          disableBackdropClick={false}
          disableEscapeKeyDown={false}
          onClose={[MockFunction]}
          open={false}
          role="dialog"
        >
          <Modal
            BackdropComponent={[Function]}
            BackdropProps={
              Object {
                "transitionDuration": Object {
                  "enter": 225,
                  "exit": 195,
                },
              }
            }
            aria-labelledby="Appearance"
            className="MuiDialog-root-9"
            classes={
              Object {
                "hidden": "MuiModal-hidden-17",
                "root": "MuiModal-root-16",
              }
            }
            disableAutoFocus={false}
            disableBackdropClick={false}
            disableEnforceFocus={false}
            disableEscapeKeyDown={false}
            disableRestoreFocus={false}
            hideBackdrop={false}
            keepMounted={false}
            manager={
              ModalManager {
                "containers": Array [],
                "data": Array [],
                "handleContainerOverflow": true,
                "hideSiblingNodes": true,
                "modals": Array [],
              }
            }
            onClose={[MockFunction]}
            open={false}
            role="dialog"
          />
        </WithStyles(Modal)>
      </Dialog>
    </WithStyles(Dialog)>
  </AppearanceDialog>
</WithStyles(AppearanceDialog)>
`;
